utils/test/CMakeLists.txt

```diff
@@ -0,0 +1,193 @@
+  testref/icecamsr2ioda.test
+  testref/icecmirs2ioda.test
+  testref/icecjpssrr2ioda.test
+  testref/insituall2ioda.test
+  testref/viirsaod2ioda.test
+)
+
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/testinput)
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/testrun)
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/obsproc/testref)
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/obsproc/testoutput)
+
+CREATE_SYMLINK( ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR}/obsproc ${obsforge_utils_test_ref} )
+CREATE_SYMLINK( ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR} ${utils_test_input} )
+
+# copy the cpp linter script
+execute_process( COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/cpplint.py ${CMAKE_BINARY_DIR}/bin/${PROJECT_NAME}_cpplint.py)
+
+# add linter for the utils
+# TODO: the linter complains about local .h files included in the .cc
+# in the rtofs directory
+# the exclude argument needs to be removed
+ecbuild_add_test( TARGET test_obsforge_util_coding_norms
+                  TYPE SCRIPT
+                  COMMAND ${CMAKE_BINARY_DIR}/bin/${PROJECT_NAME}_cpplint.py
+                    ARGS
+                    --filter=-legal/copyright
+                    --linelength=120
+                    --quiet
+                    --recursive
+                    --exclude=${CMAKE_CURRENT_SOURCE_DIR}/../obsproc/rtofs/*.cc
+                    ${CMAKE_CURRENT_SOURCE_DIR}/../
+                  WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/bin )
+
+# Prepare data for the IODA converters
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/obsproc)
+ecbuild_add_test( TARGET  test_obsforge_util_prepdata
+                  COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/prepdata.sh
+                  ARGS    ${CMAKE_CURRENT_SOURCE_DIR}
+                  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/obsproc)
+
+# Test the RADS to IODA converter
+ecbuild_add_test( TARGET  test_obsforge_util_rads2ioda
+                  COMMAND ${CMAKE_BINARY_DIR}/bin/obsforge_obsprovider2ioda.x
+                  ARGS    "../testinput/obsforge_rads2ioda.yaml"
+                  LIBS    netcdf2ioda
+                  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/obsproc
+                  TEST_DEPENDS test_obsforge_util_prepdata)
+
+# Test the GHRSST to IODA converter
+ecbuild_add_test( TARGET  test_obsforge_util_ghrsst2ioda
+                  COMMAND ${CMAKE_BINARY_DIR}/bin/obsforge_obsprovider2ioda.x
+                  ARGS    "../testinput/obsforge_ghrsst2ioda.yaml"
+                  LIBS    obsforge-utils
+                  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/obsproc
+                  TEST_DEPENDS test_obsforge_util_prepdata)
+
+# copy rtofs binary input files to the testing area
+# and generate the tests
+
+set(RTOFS_INPUT_FILE1 "rtofsinsitu_2024032600.profile")
+set(RTOFS_INPUT_FILE2 "rtofsinsitu_2024032700.profile")
+
+# rtofs directory on orion:
+set(RTOFS_FILES_PATH
+	"/work/noaa/da/marineda/gfs-marine/data/obs/ci/obs"
```

Comment:
This should be set by env variable in the modulefiles

Reply:
Thanks @CoryMartin-NOAA , created an issue for this.